cluster-autoscaler/cloudprovider/aws/aws_wrapper.go

```diff
@@ -107,59 +109,49 @@ func (m *awsWrapper) getAutoscalingGroupsByNames(names []string) ([]*autoscaling
 	return asgs, nil
 }
 
-func (m *awsWrapper) getAutoscalingGroupNamesByTags(kvs map[string]string) ([]string, error) {
-	// DescribeTags does an OR query when multiple filters on different tags are
-	// specified. In other words, DescribeTags returns [asg1, asg1] for keys
-	// [t1, t2] when there's only one asg tagged both t1 and t2.
+func (m *awsWrapper) getAutoScalingGroupsByTags(configs []asgAutoDiscoveryConfig) ([]*autoscaling.Group, error) {
+	if len(configs) == 0 {
+		return nil, nil
+	}
+
 	filters := []*autoscaling.Filter{}
-	for key, value := range kvs {
-		filter := &autoscaling.Filter{
-			Name:   aws.String("key"),
-			Values: []*string{aws.String(key)},
-		}
-		filters = append(filters, filter)
-		if value != "" {
+	tagNames := []string{}
+	for _, config := range configs {
+		for key, value := range config.Tags {
 			filters = append(filters, &autoscaling.Filter{
-				Name:   aws.String("value"),
-				Values: []*string{aws.String(value)},
+				Name:   aws.String("tag-key"),
+				Values: []*string{aws.String(key)},
 			})
+			if value != "" {
+				filters = append(filters, &autoscaling.Filter{
+					Name:   aws.String("tag-value"),
+					Values: []*string{aws.String(value)},
+				})
+			}
+			tagNames = append(tagNames, key)
 		}
 	}
+	klog.V(4).Infof("Regenerating instance-to-ASG map for auto-discovery tags: %v", tagNames)
 
-	tags := []*autoscaling.TagDescription{}
-	input := &autoscaling.DescribeTagsInput{
+	asgs := make([]*autoscaling.Group, 0)
+
+	input := &autoscaling.DescribeAutoScalingGroupsInput{
 		Filters:    filters,
 		MaxRecords: aws.Int64(maxRecordsReturnedByAPI),
 	}
 	start := time.Now()
-	err := m.DescribeTagsPages(input, func(out *autoscaling.DescribeTagsOutput, _ bool) bool {
-		tags = append(tags, out.Tags...)
+	err := m.DescribeAutoScalingGroupsPages(input, func(output *autoscaling.DescribeAutoScalingGroupsOutput, _ bool) bool {
+		asgs = append(asgs, output.AutoScalingGroups...)
 		// We return true while we want to be called with the next page of
 		// results, if any.
 		return true
 	})
-	observeAWSRequest("DescribeTagsPages", err, start)
```

Comment:
is DescribeTagsPages used elsewhere in CA or can we get rid of this permission now?

Reply:
It doesn't look like it's used elsewhere. Will remove this permission.